Display projects in alphabetical order

andrew-js-wright opened the pull request:
Currently when adding a project to a technology the projects are listed in the
order they were retrieved from the database. In order to aid user experience the
projects should be ordered sensibly.

Connects to #102

andrew-js-wright committed:
Display projects in alphabetical order

Currently when adding a project to a technology the projects are listed in the
order they were retrieved from the database. In order to aid user experience the
projects should be ordered sensibly.

Connected to #102

andrew-js-wright commented:
#102

willh commented:
Are there any other places we could or should default the ordering of projects to alphabetical than the tech page sidebar? If yes, what about setting a default ordering in the projects DAO when fetching project records and not sorting in-memory once they're sent to the frontend?
https://github.com/KainosSoftwareLtd/TechRadar/blob/a42b187fd314d5e5ad07ca6286eb9c77b4f1c926/dao/projects.js#L11

willh reviewed (commented):
Update to tests ensures that when getAll method is called on project with the stub array that it'll not fail due to missing mocked methods, but it doesn't create a situation where the tests would fail if the ordering either didn't happen or was wrong. If I moved the `return 0` line up by two the sorting wouldn't happen but the test wouldn't fail. Can we add a test that asserts that the result of the getAll function is an appropriately sorted list/array?

andrew-js-wright commented:
Good point Will, the only place `project.getAll` is called from is the technology helper so this ordering could be moved to the DAO method.

andrew-js-wright commented:
Moved sorting to DAO layer. This is consistent to how ordering is done on the other objects. Reverted other changes. There exists no tests for the project DAO. While this is a bad thing I don't feel that this change justifies writing a new test class.

willh commented:
> Good point Will, the only place project.getAll is called from is the technology helper so this ordering could be moved to the DAO method.

I think the thing that would swing this for me isn't just that it is only called from that one place, but that we don't reasonably foresee having to undo that (or implement a different order) in some other place in the future, by setting this as a new expected default. 

Also I agree with risk/rewards tradeoff on test decision for adding an `ORDER BY` to an existing query - we might change our minds about that once we're doing something more interesting than ordered straight selects in the data layer!
